### ISSUE TITLE
Page in Backend List is not applied correctly after filterChange

### ISSUE THREAD

**Issue by LordRazen** [octobercms/october#5772]
OctoberCMS 3.5.13

I tested the behaviour in the DemoPlugin / Locations:

Bug: Basically the page goes to page 1 if you change the filter but the page is reapplied if you refresh again.

- Change `recordsPerPage: 1` within the config_list.yaml in october/test/controllers/locations
- Go to Page 2
- Switch Country Filter to "Petoria" (fox ex) => Page is switched to Page 1
- Press F5 => Page is switched to Page 2

Basically that's cause the page property is not cleaned from Input:all() if you switch the filter. It's still there. 

Expectations: 
- If you change the filter, the page should start at page 1, but this should also be page 1 if refresh is used.

**Comment by daftspunk** [comment]
This is fixed in v3.6

Thanks!
